build/common.sh

```diff
@@ -472,11 +488,8 @@ function kube::build::build_image() {
   kube::version::save_version_vars "${build_context_dir}/kube-version-defs"
 
   cp build/build-image/Dockerfile ${build_context_dir}/Dockerfile
-  if kube::build::is_osx; then
-    sed -i "" "s/KUBE_BUILD_IMAGE_CROSS/${KUBE_BUILD_IMAGE_CROSS}/" ${build_context_dir}/Dockerfile
```

Comment:
yeah, the original `sed` fix was in https://github.com/kubernetes/kubernetes/pull/20147. I don't know whether to blame OSX/BSD or GNU for being different.

In any case, this PR cleans it up a bit.